power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -330,24 +336,109 @@ template <bool sym> class YBus {
         auto const& y_bus_entry_indptr = y_bus_struct_->y_bus_entry_indptr;
         // loop for each y bus position
         for (Idx entry = 0; entry != nnz(); ++entry) {
+            // start admittance accumulation with zero
+            ComplexTensor<sym> entry_admittance{0.0};
+            IdxVector entry_param_shunt;
+            IdxVector entry_param_branch;
+            // std::vector<std::pair<Idx, Idx>> entry_param_shunt_pair;
+            //  loop over all entries of this position
+            for (Idx element = y_bus_entry_indptr[entry]; element != y_bus_entry_indptr[entry + 1]; ++element) {
+                auto param_idx = y_bus_element[element].idx;
+                if (y_bus_element[element].element_type == YBusElementType::shunt) {
+                    // shunt
+                    entry_admittance += shunt_param[param_idx];
+                    entry_param_shunt.push_back(param_idx);
+                } else {
+                    // branch
+                    entry_admittance +=
+                        branch_param[param_idx].value[static_cast<Idx>(y_bus_element[element].element_type)];
+                    entry_param_branch.push_back(param_idx);
+                    // entry_param_shunt_pair.push_back(std::make_pair(param_idx, element));
+                }
+            }
+            // assign
+            admittance[entry] = entry_admittance;
+            map_admittance_param_branch.push_back(entry_param_branch);
+            map_admittance_param_shunt.push_back(entry_param_shunt);
+            // map_admittance_param_shunt_pairs.push_back(entry_param_shunt_pair);
+        }
+        // move to shared ownership
+        admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
+    }
+
+    IdxVector increments_to_entries() {
+        // construct affected entries
+        std::set<Idx> affected_entries;
+        auto querry_params_in_map = [&affected_entries, this](auto params_to_change, bool is_branch) {
```

Comment:
`auto` uses a copy. should be `const auto&`

```suggestion
        auto querry_params_in_map = [&affected_entries, this](auto const& params_to_change, bool is_branch) {
```

Reply:
I hope compilers in the future could infer `const` into `auto` as well